web-services/query/src/main/java/datawave/webservice/query/logic/BaseQueryLogic.java

```diff
@@ -26,6 +26,7 @@
     private GenericQueryConfiguration baseConfig;
     private String logicName = "No logicName was set";
     private String logicDescription = "Not configured";
+    protected String queryPlan = "No Query Plan was set.";
```

Comment:
I guess the point I was trying to make is that if we can set this value in the configuration, then we do not want it also set here.  We used to have almost all of the configuration both in the logic and in the configuration which was causing a nightmare in terms of ensure values got set in both places.  Lets have this set only in the configuration if at all possible or give me a good reason why.

Reply:
Ah okay, I see what you were saying now. The reason I set this in the query logic is because when a query plan fails it returns back the configuration as null, which is why it can't be sent to metrics. To only set it in the configuration would require to handle the exceptions within the configuration and still return the configuration when the query plan fails. Is that preferable?

Reply:
In the RunningQuery where you handle the exception case, you use logic.getQueryPlan().  There is no reason why logic.getQueryPlan() couldn't get the plan from the config object instead of from a local class member.  If you look at almost all of the other setters and getters in the logic, they simply defer to the configuration object.

Reply:
I believe I just made this change.